includes/admin/class-sensei-setup-wizard.php

```diff
@@ -189,18 +189,20 @@ public function prepare_wizard_page() {
 	public function activation_redirect() {
 		if (
 			// Check if activation redirect is needed.
-			! get_transient( 'sensei_activation_redirect' )
+			! get_option( 'sensei_activation_redirect', false )
 			// Test whether the context of execution comes from async action scheduler.
 			// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
 			|| ( isset( $_REQUEST['action'] ) && 'as_async_request_queue_runner' === $_REQUEST['action'] )
 			// On these pages, or during these events, postpone the redirect.
-			|| wp_doing_ajax() || wp_doing_cron() || is_network_admin() || ! current_user_can( 'manage_sensei' )
+			|| wp_doing_ajax() || wp_doing_cron() || is_network_admin()
+			// Only redirects for admin users.
+			|| ! current_user_can( 'manage_sensei' )
+			// Check if it's an admin screen that should redirect.
+			|| ! $this->should_current_page_redirect_to_wizard()
```

Comment:
Now the redirect runs when the user accesses the pages we used to display the Setup Wizard notice (Dashboard, Plugins, and Sensei pages).